helm/cluster-vsphere/templates/ipam/assign-ip-pre-install-job.yaml

```diff
@@ -96,10 +96,32 @@ spec:
               # patch the vspherecluster.spec.controlPlaneEndpoint.host
               kubectl patch vspherecluster -n {{ $.Release.Namespace }} {{ include "resource.default.name" $ }} --type=merge -p '{"spec":{"controlPlaneEndpoint":{"host": "'${new_ip}'"}}}'
 
+              PROVIDER="{{ $.Values.cluster.providerIntegration.resourcesApi.controlPlaneResource.provider }}"
```

Comment:
This section adds logic to patch the `kubeadmcontrolplane` or `kamajicontrolplane` CR based on the value set in `.cluster.providerIntegration.resourcesApi.controlPlaneResource.provider `.